### PR TITLE
Volumes view

### PR DESCRIPTION
**Pull request by TristanWright**
Definitely work in progress

We can save and delete but not attach right now.

### CODE REVIEW

File: src/pages/Preferences/Volumes/VolumeForm.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import React              from 'react';
+import deepClone          from 'mout/src/lang/deepClone';
+import style              from 'HPCCloudStyle/ItemEditor.mcss';
+
+// const types = {
```
Comment (TristanWright) [inline src/pages/Preferences/Volumes/VolumeForm.js]: right now cumulus just takes the type 'ebs'

I tried working this in but commented it out for later.

File: src/pages/Preferences/Volumes/VolumeForm.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                  <option
+                    key={ `${prof._id}` }
+                    value={ prof._id }
+                  >
+                  { prof.name }
+                  </option>
+                )}
+              </select>
+          </section>
+          { this.props.data._id ?
```
Comment (TristanWright) [inline src/pages/Preferences/Volumes/VolumeForm.js]: The attach to cluster form does not appear unless volume is saved

File: src/workflows/nwchem/common/steps/Simulation/Start/index.js [hunk 3]
```diff
@@ -34,7 +34,6 @@ const SimulationStart = React.createClass({
       serverType: 'Traditional',
       EC2: defaultServerParameters.EC2,
       Traditional: defaultServerParameters.Traditional,
-      OpenStack: defaultServerParameters.OpenStack,
```
Comment (TristanWright) [inline src/workflows/nwchem/common/steps/Simulation/Start/index.js]: should probably revert this if we're going to use open stack in the long run?

File: style/PageWithMenu.mcss [hunk 4]
```diff
@@ -122,3 +122,15 @@
   composes: fa-spin    from 'font-awesome/css/font-awesome.css';
   margin-right: 5px;
 }
+
```
Comment (TristanWright) [inline style/PageWithMenu.mcss]: for the attach and detach icons we're using http://fontawesome.io/icons/ forward and reply, an attached volume in the to-be-created status view will have use the paperclip icon

**Comment by TristanWright** [comment]
@cjh1 this is ready fro you to try out.

**Comment by cjh1** [comment]
I am trying to test this PR out and am getting the following error?

```
HPCCloud.js:58762 Uncaught (in promise) TypeError: Cannot read property '_id' of undefined
    at Object.updateItem (HPCCloud.js:58762)
    at projectsReducer (HPCCloud.js:58605)
    at combination (HPCCloud.js:30074)
    at dispatch (HPCCloud.js:29621)
    at Object.dispatch (HPCCloud.js:54836)
    at dispatch (HPCCloud.js:54882)
    at HPCCloud.js:57182
updateItem @ HPCCloud.js:58762
projectsReducer @ HPCCloud.js:58605
combination @ HPCCloud.js:30074
dispatch @ HPCCloud.js:29621
(anonymous) @ HPCCloud.js:54836
dispatch @ HPCCloud.js:54882
(anonymous) @ HPCCloud.js:57182
```

**Comment by TristanWright** [comment]
What page is this error happening on?

**Comment by cjh1** [comment]
When I try to create a project

**Comment by TristanWright** [comment]
So this is from a mistake I made when doing #583.

#585 fixes it

**Comment by TristanWright** [comment]
I can run through a pyfr and a visualization job. Having a ws with the final step of actually visualizing though, something with the visualizer itself I think. 👍

**Comment by cjh1** [comment]
@TristanWright Awesome! Thanks for continuing to push on this :medal_sports:

**Comment by cjh1** [comment]
I just tried to test this and am running into a problem with the PyFR workflow, I am seeing the following celery error:

```python
[2017-02-24 21:40:11,052: ERROR/MainProcess] Received unregistered task of type u'hpccloud.taskflow.pyfr.setup_input'.
The message has been ignored and discarded.
```

It looks like the PyFR taskflow is not being picked up. I know @jourdain did some restructure of the taskflow directory we may been to rebase on master to see if it fixed there.

**Comment by cjh1** [comment]
Looks like its up to date, in which case this is probably broken in master. Which workflow where you using to test this branch ParaViewWeb?

**Comment by cjh1** [comment]
@jourdain Wasn't there a fix for this?

**Comment by TristanWright** [comment]
I I tested this on a PyFR workflow, I've never seen that error though.

**Comment by jourdain** [comment]
I've tested all workflow on traditional cluster and everything was fine.

I think there was a fix I pushed in cumulus to pick modules as well as files. But that's just what I remember.

**Comment by cjh1** [comment]
Ok, I figured it out, need to rebase cumulus branch on master, the fix was in cumulus.

**Comment by cjh1** [comment]
Now I am seeing;

```python

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "cumulus/taskflow/__init__.py", line 117, in wrapped
    return func(celery_task, *args, **kwargs)
  File "/opt/hpccloud/hpccloud/server/taskflows/hpccloud/taskflow/pyfr/__init__.py", line 185, in setup_input
    mesh_file_id = kwargs['input']['meshFile']['id']
KeyError: 'id'

```
Looks like the mesh file id is not being passed to the taskflow. 

These are the SHA's I am using:
cumulus: 49cda2f2
hpccloud: a82e4781

I will do some more digging next week.

**Comment by cjh1** [comment]
I was able to run a PyFR workflow to completion. However, no volume was create or mounted?

**Comment by cjh1** [comment]
@TristanWright There seem to be two volume size fields?

**Comment by cjh1** [comment]
I'm not seeing any log messages show up for the attach_volume(...) task.

**Comment by cjh1** [comment]
OK, so I am now seeing the volume being attached and mounted on the head node at /data :smile: The final step is to get the simulation result written to this volume, for this we need to do two things:

- [ ]  Export /data to other node in the cluster using NFS
- [ ] Update the job directory base to point to /data

@TristanWright If you would like I can take a look at adding these last two parts?

**Comment by TristanWright** [comment]
> @TristanWright There seem to be two volume size fields?

The first one is definitely redundant, I'll remove that.

> we need to do two things:
 Export /data to other node in the cluster using NFS
 Update the job directory base to point to /data
@TristanWright If you would like I can take a look at adding these last two parts?

I thought it was being uploaded to /data (or wait are we [just mounting it at /data](https://github.com/Kitware/cumulus/pull/297/files#diff-f2187e7e6ad7bfc29cbcd7181b27b676R293))? Yeah that would wrap up this branch while I'm working on sharing.

**Pull request merged**